### ISSUE TITLE
HOW-TO: Install RHEL 6.5 

### ISSUE THREAD

**Issue by WanpengQian** [churchers/vm-bhyve#50]
I think Linux distribution is a mess.... so many distributions.
and since bhyve does't support vga output yet, it is better to have some HOW-TOs to particular Linux Distributions.
and here is some tips on How to install Redhat Enterprise Linux 6.5 under vm-bhyve.

**Comment by WanpengQian** [comment]
the file name of iso is ```rhel-server-6.5-x86_64-dvd.iso```,
and the default kernel of rhel 6.5 is ```2.6.32-431.el6.x86_64```

here is my vm config file.
```
guest="centos"
linux_kernel="2.6.32-431.el6.x86_64"
cpu=8
memory=64G
network0_type="virtio-net"
network0_switch="public"
disk0_type="virtio-blk"
disk0_name="disk0"
disk0_dev="sparse-zvol"
uuid="72f33ef7-ba87-11e5-b42a-00199982f742"
network0_mac="58:9c:fc:00:c6:4a"
```

in order to boot after CD installation, we have to modify vm-guest line 169
```
__log_and_write "write" "${_name}" "boot.cmd" "linux /vmlinuz-${_kernel} root=/dev/mapper/VolGroup-lv_root"
```

after that , you can start installation by ``` vm install GUESTNAME rhel-server-6.5-x86_64-dvd.iso```

since rhel 6.5 already has serial output, ```vm console  GUESTNAME ``` will bring you to text console.

that is all.

**Comment by churchers** [comment]
Thanks for providing the useful information for people wanting to run this OS.

Linux support in bhyve at the moment is a bit of a nightmare. It needs rewriting in vm-bhyve ideally. I'm hoping that UEFI (and hopefully VGA support) will get to the point where I can just let the guests treat bhyve as a standard system, rather than having to constantly tweak grub settings for everything.

**Comment by mrqwer88** [comment]
Hello!

You have plannes for move root path and  /boot/ path to vm config?
It not very comfortable to fix it every time in vm-guest file and it can be not equal, if we have few vm-s with centos.
As example my actual vm need this boot.cmd - 
```
linux /boot/vmlinuz-2.6.32-573.18.1.el6.x86_64 root=/dev/vda1
initrd /boot/initramfs-2.6.32-573.18.1.el6.x86_64.img
boot
```
(it have not LVM - only one part with / and not separate /boot/ partition)

**Comment by pavel-odintsov** [comment]
Same question. Do you have some plans to unify this approach?

**Comment by churchers** [comment]
All grub configuration has been moved to the configuration files in `0.10`.
Theoretically it should be possible to boot anything that is supported by `grub-bhyve`/`bhyve`. The only thing I currently don't support is specifying values for the grub-bhyve `--directory` and `--grub-cfg` options, which I will probably add soon just in case anyone has a use for them as it's a fairly small change.

See the GitHub readme and sample templates for details and examples. If the commands in the reply above boot the guest, it should be possible to just change/add the following in guest configuration

    guest="linux"
    grub_run0="linux /boot/vmlinuz-2.6.32-573.18.1.el6.x86_64 root=/dev/vda1"
    grub_run1="initrd /boot/initramfs-2.6.32-573.18.1.el6.x86_64.img"
    grub_run2="boot"

If the partition needs to be specified (which I've found is required in most of my tests), either of the following is functional:

    grub_run0="linux (hd0,msdos1)/boot/..."

or 

    grub_run_partition="msdos1"
    grub_run0="linux /boot/..."
